pkg/gitprovider/gitness.go

```diff
@@ -281,3 +296,76 @@ func (g *GitnessGitProvider) GetDefaultBranch(staticContext *StaticGitContext) (
 	client := g.getApiClient()
 	return client.GetDefaultBranch(staticContext.Url)
 }
+
+func (g *GitnessGitProvider) RegisterPrebuildWebhook(repo *GitRepository, endpointUrl string) (string, error) {
+	client := g.getApiClient()
+	webhook, err := client.CreateWebhook(repo.Id, repo.Owner, gitnessclient.Webhook{
+		Triggers:    []string{"branch_updated"},
+		Url:         endpointUrl,
+		Identifier:  "daytona-webhook_" + repo.Id,
+		DisplayName: "Daytona Webhook",
+		Enabled:     true,
+	})
+	if err != nil {
+		return "", err
+	}
+	return webhook.Uid, nil
+}
+
+func (g *GitnessGitProvider) GetPrebuildWebhook(repo *GitRepository, endpointUrl string) (*string, error) {
+	client := g.getApiClient()
+	webhooks, err := client.GetAllWebhooks(repo.Id, repo.Owner)
+	if err != nil {
+		return nil, err
+	}
+	for _, webhook := range webhooks {
+		if webhook.Url == endpointUrl {
+			return &webhook.Uid, nil
+		}
+	}
+	return nil, nil
+}
+
+func (g *GitnessGitProvider) UnregisterPrebuildWebhook(repo *GitRepository, id string) error {
+	client := g.getApiClient()
+	return client.DeleteWebhook(repo.Id, repo.Owner, id)
+}
+
+func (g *GitnessGitProvider) GetCommitsRange(repo *GitRepository, initialSha string, currentSha string) (int, error) {
```

Comment:
This method doesn't make sense 😅 

The aim of the method is to get the number of commits between to SHAs. 
In this PR, triggering a prebuild by committing to the repo doesn't work.

When you set commit interval 1 on the prebuild, it should trigger on all commits. You need to test that.

Reply:
You don't count the initial sha.

If the commits go like this
```
initialSha
sha2
sha3
sha4
currentSha
```

the functions needs to return 4.

Reply:
It should not trigger a new build if the commit interval was not exceeded.

To test trigger files, leave the commit interval on 10. If you set it on 1, a new build will always trigger.

Reply:
You can think of the `Commit interval` and `Trigger files` as conditions with an `OR` (`||`) between them. So if one of the conditions is met, then the prebuild should trigger.

Reply:
@the-johnwick this method doesn't seem correct again because it doesn't even use `currentSha`.
From what I can see, you simply list all commits starting from `initialSha` but you need to list them up until `currentSha`.

Take a look at how other providers implement this function. e.g. https://github.com/daytonaio/daytona/blob/main/pkg/gitprovider/github.go#L446